src/main/java/net/tridentsdk/chat/ChatComponent.java

```diff
@@ -0,0 +1,482 @@
+    public ChatComponent addExtra(ChatComponent component) {
+        if (!hasExtra(component, true)) {
+            $lock.writeLock().lock();
+            this.extra.add(component);
+            $lock.writeLock().unlock();
+        }
+        return this;
+    }
+
+    /**
+     * Adds a string to the 'extra' array.
+     *
+     * @param string The string.
+     *
+     * @return This component.
+     */
+    public ChatComponent addExtra(String string) {
+        $lock.writeLock().lock();
+        this.extra.add(ChatComponent.create().setText(string));
+        $lock.writeLock().unlock();
+        return this;
+    }
+
+    /**
+     * Checks if this component contains the given component, optionally recursively.
+     *
+     * @param component The component
+     * @param recursive Whether to check children as well.
+     *
+     * @return True iff the given component exists in this component's heirarchy.
+     */
+    public boolean hasExtra(ChatComponent component, boolean recursive) {
+        $lock.readLock().lock();
+        if (extra.contains(component)) {
+            $lock.readLock().unlock();
+            return true;
+        }
+        if (!recursive) {
+            $lock.readLock().unlock();
+            return false;
+        }
+        for (ChatComponent child : extra) {
+            if (child.hasExtra(component, true)) {
+                $lock.readLock().unlock();
+                return true;
+            }
+        }
+        $lock.readLock().unlock();
+        return false;
+    }
+
+    /**
+     * Gets whether or not this component is bold.
+     *
+     * @return True iff it is.
+     */
+    public boolean isBold() {
+        return bold != null && bold.get();
+    }
+
+    /**
+     * Sets this component's boldness.
+     *
+     * @param bold The boldness.
+     *
+     * @return This component.
+     */
+    public ChatComponent setBold(boolean bold) {
+        $lock.writeLock().lock();
+        if (this.bold == null)
+            this.bold = new AtomicBoolean(bold);
+        else
+            this.bold.set(bold);
+        $lock.writeLock().unlock();
+        return this;
+    }
+
+    /**
+     * Gets whether or not this component is in italics.
+     *
+     * @return True iff it is.
+     */
+    public boolean isItalic() {
+        return italic != null && italic.get();
+    }
+
+    /**
+     * Sets this component's italic.
+     *
+     * @param italic The italic.
+     *
+     * @return This component.
+     */
+    public ChatComponent setItalic(boolean italic) {
+        $lock.writeLock().lock();
+        if (this.italic == null)
+            this.italic = new AtomicBoolean(italic);
+        else
+            this.italic.set(italic);
+        $lock.writeLock().unlock();
+        return this;
+    }
+
+    /**
+     * Gets whether or not this component is underlined.
+     *
+     * @return True iff it is.
+     */
+    public boolean isUnderlined() {
+        return underlined != null && underlined.get();
```

Comment:
THREAD-SAFETY
This is a poor construct to use for a thread-safe class. It doesn't save (that much) memory, and can be solved by using a volatile boolean set to false as a default value instead.